### PR TITLE
[C-API] Bugfix for ml_pipeline_element_get_handle

### PR DESCRIPTION
**Pull request by songgot**
[C-API] Bugfix for ml_pipeline_element_get_handle
    
    Need to ref element obtained by iterate_element() and unref it in cleanup_node() for common_elem
    
    - common_elem by ml_pipeline_element_get_handle() need to unref in cleaunup_node() because g_hash_table_lookup assign new memory.
      but normal operation, unref is not necessary, so add ref in iterate_element()
    - Now, the element's finalize function is called after ml_pipeline_element_get_handle()

### CODE REVIEW

**Comment by jaeyun-jung** [comment]
This may cause an error in iterate_element() function.
The function iterate_element() internally finds elements in pipeline and adds ml_pipeline_element into the table.
I recommend decreasing the ref-count of gst-element in cleanup_node() function.

iterate_element()
```
e = construct_element (elem, pipe_h, name, element_type);
>> change this line
e = construct_element (gst_object_ref (elem), pipe_h, name, element_type);
```

cleanup_node()
```
  if (e->src)
    gst_object_unref (e->src);
  if (e->sink)
    gst_object_unref (e->sink);

  gst_object_unref (e->element);  << add this
  _ml_tensors_info_free (&e->tensors_info);
```

**Comment by songgot** [comment]
Thanks for the detailed explanation offline.

File: c/src/ml-api-inference-pipeline.c [hunk 1]
```diff
@@ -601,6 +601,8 @@ cleanup_node (gpointer data)
   if (e->sink)
     gst_object_unref (e->sink);
 
+  g_object_unref (e->element);
```
Comment (jaeyun-jung) [inline c/src/ml-api-inference-pipeline.c:604]: maybe gst_object_unref()

Comment (songgot) [reply c/src/ml-api-inference-pipeline.c:604]: Thank you!!

File: c/src/ml-api-inference-pipeline.c [hunk 2]
```diff
@@ -915,7 +917,8 @@ iterate_element (ml_pipeline * pipe_h, GstElement * pipeline,
               if (element_type != ML_PIPELINE_ELEMENT_UNKNOWN) {
                 ml_pipeline_element *e;
 
-                e = construct_element (elem, pipe_h, name, element_type);
+                e = construct_element (gst_object_ref (elem), pipe_h, name,
```
Comment (jaeyun-jung) [inline c/src/ml-api-inference-pipeline.c:920]: let's consider failure case if construct_element() returns null, at line 930 - allocation failure,
```
} else {
  /* allocation failure */
  gst_object_unref (elem);
```

Comment (songgot) [reply c/src/ml-api-inference-pipeline.c:920]: Okay, I will add it.

**Review by jaeyun-jung — APPROVED** [review]

**Review by niley7464 — APPROVED** [review]
LGTM!

**Review by anyj0527 — APPROVED** [review]
LGTM. Thanks!

**Comment by songgot** [comment]
Thank you for review

**Pull request merged**